analysis_driver/util/helper_functions.py

```diff
@@ -0,0 +1,6 @@
+# retrieving run_element_id, which differs between barcodeless and multiplexed runs
+def get_run_element_id(run_id, lane_number, barcode=None):
+    if barcode is None:
+        return '{dataset_name}_{lane}'.format(dataset_name=run_id, lane=lane_number)
+    else:
+        return '{dataset_name}_{lane}_{barcode}'.format(dataset_name=run_id, lane=lane_number, barcode=barcode)
```

Comment:
It's annoying that we need to do this, but it all works fine

Reply:
@mwhamgenomics I agree, that's why I placed it in a helper function so at least it's only found once in the source code.